File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    // participant와 completion을 sort한다.
    participant.sort();
    completion.sort();
    
    // 만약 participant[i]와 completion[i]가 다르면 해당 이름을 return한다.
    for(var i = 0; i < participant.length; i++) {
        if(participant[i] !== completion[i]) {
            return participant[i];
        }
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    
    dict = {}
    
    for person in participant:
        if person in dict:
            dict[person] += 1
        else:
            dict[person] = 1
    
    
    for person in completion:
        dict[person] -= 1
        
    for person, count in dict.items():
        if count > 0 :
            return person